### DISCUSSION TITLE
Would Fontsource handle fonts having inconsistent weights/styles?

### DISCUSSION THREAD

**Discussion by thasmo** [fontsource/fontsource#386] [General]
I've noticed that [`Open Sans Condensed`](https://fonts.google.com/specimen/Open+Sans+Condensed) has three font faces:

* `300 normal`
* `300 italic`
* `700 normal`

It does not feature a `700 italic` face.

The distributed `metadata.json` file has this properties:
```
  "weights": [300, 700],
  "styles": ["italic", "normal"],
```
Which do not and can not indicate that the `700 italic` face is not available at all.

Also, the `mixins.scss` file has this defined:
```
$weight: 400;
```
Which is probably wrong, because `Open Sans Condensed` does not feature a weight of `400`.

Are these known issues? Regardless, is there a need or goal for `Fontsource` to provide such granularity for the distributed `npm` packages?

**Comment by ayuhito** [comment]
It is a known issue and there have been fixes in the past to address CSS files generating without a font file due to this wonky matchup.

`metadata.json` was left as is since I wasn't too sure how to address this concern. If users really needed to figure out the difference, we could introduce an array that would show something like this `[300,300i,700]`, but is there actually any demand for that? Hence, I just never addressed that aspect.

Yeah, we never checked the default value for mixins, but it's a very easy fix since we already have a utility function that finds the nearest available weight [here](https://github.com/fontsource/fontsource/blob/main/scripts/utils/utils.ts#L45).
